thesis/inc/technical/detailed_architecture/meta_information.tex

```diff
@@ -0,0 +1,60 @@
+\subsection{Meta Information}
+\label{subsec:detailed_meta_information}
+The meta information structure is an important structure in NOX ECS,
+it contains all type information needed by the ECS to work without
+templates and virtual functions.
+Each component type has a meta information, containing information
+of the components size, type identifier\secref{subsec:detailed_type_identifier},
+operation types\secref{subsec:detailed_operation_types}, data access dependencies etc.
+All of this information allows NOX ECS to implement type ignorant generic functionality.
+
+\subsubsection{Explanation}
+Meta information is simply a struct holding data relevant to a type.
+It is stored together with a component collection\secref{subsec:detailed_component_collection},
+which uses the information stored in the meta structure to decide what actions
+it should take on the components stored within itself.
+Meaning that the meta information both acts as a replacement for templates,
+and for virtual tables.
+
+Meta information objects are created through a dedicated template function,
+which uses the template argument to create a meta information structure,
+containing type information, as well as wrappers for the different operation types.
+
+\subsubsection{Motivation}
+The meta information structure was mainly motivated by the requirement of avoiding heavy template and virtual functionality\footnote{Detailed in sections \ref{sec:requirements_templates} and \ref{subsec:requirements_performance_virtual_functions}}.
+NOX ECS would need a way to store general type information that could be use to write generic functionality, without templates,
```

Comment:
Omit general, isn't "type information" enough? Also it should say "that could be used to write"